app/controllers/posts_controller.rb

```diff
@@ -48,8 +46,35 @@ def destroy
     end
   end
 
+  def api_index
+    @user = User.find(params[:user_id])
+    @posts = Post.where(author_id: @user.id).includes(:comments, :user).paginate(page: params[:page], per_page: 10)
+    render json: @posts
+  end
+
+  def api_comments
+    @post = Post.find(params[:post_id])
+    @comments = @post.comments
+    render json: @comments
+  end
+
+  def api_add_comment
+    @post = Post.find(params[:post_id])
+    @comment = Comment.new(comment_params)
+    @comment.user = current_user
+    if @comment.save
+      render json: @comment, status: :created
+    else
+      render json: { error: 'Failed to create comment' }, status: :unprocessable_entity
+    end
+  end
```

Comment:
- [ ] When I try to access any of the endpoints routed to these actions in the `Posts` controller, I get this issue:

![image](https://github.com/maths29/blog_app/assets/77106962/7f8cbdba-bc3f-4fe3-a162-dddc576fc947)

---

This issue is a result of CanCan denying access to the API endpoints. In my opinion, I would advise that you create your endpoints in this way to avoid issues with the UI side of the application:

- **GET :** `/api/users/:user_id/posts` or `/api/v1/users/:user_id/posts`
- **GET :** `/users/:user_id/posts/:id/comments`
- **POST :** `/users/:user_id/posts/:id/comments`

You can read this [article](https://medium.com/nyc-ruby-on-rails/versioning-apis-ruby-on-rails-design-patterns-b0702b93d96b) to have a better understanding of what needs to be done. 👍 🖱️